templates/rocket.service.erb

```diff
@@ -13,7 +13,7 @@ RestartSec=10
 StandardOutput=syslog
 StandardError=syslog
 SyslogIdentifier=rocket.chat
-Environment=INSTANCE_IP=<%= @instance_ip || scope['::facts']['networking']['ip'] %>
+Environment=INSTANCE_IP=<%= @instance_ip || scope.lookupvar('networking::ip') %>
```

Comment:
This form, `scope.lookupvar('networking::ip')`, does not work, at least on Puppet 6.0.4. What was wrong with the correct `scope['::facts']['networking']['ip']`, and how did you test your change?

If `scope['facts']` is a problem, then maybe we can compromise on `scope.lookupvar('facts')`?

Reply:
If, on some old version of Puppet, one of `scope['::facts']` and `scope['::facts']['networking']` is `nil`, then the right compromise would be on the legacy fact, which may simply be called `@ipaddress` in the ERB template.